QuizApp/QuizSession/QuizSessionView.swift

```diff
@@ -0,0 +1,78 @@
+import SwiftUI
+import Factory
+
+struct QuizSessionView: View {
+
+    @ObservedObject var viewModel: QuizSessionViewModel
+
+    @State private var numberOfCorrectQuestions = 0
+    @State private var progressColors: [Color]
+
+    init(viewModel: QuizSessionViewModel) {
+        self.viewModel = viewModel
+        progressColors = [Color](repeating: Color.white.opacity(0.3), count: viewModel.quiz.numberOfQuestions)
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack {
+                ProgressView(
+                    progressColors: $progressColors,
+                    numberOfQuestions: viewModel.quiz.numberOfQuestions)
+
+                VStack(alignment: .leading) {
+                    if viewModel.currentQuestion != nil {
```

Comment:
Why would the current question ever be `nil`?

Reply:
It's ```nil```in the beginning until all the questions are fetched (app crashed without ```if viewModel.currentQuestion != nil```).